core/utils/path.ts

```diff
@@ -39,9 +34,9 @@ export function getPath(...parts: string[]) {
 
 const staticUrl = new URL(import.meta.resolve("../../static/"));
 
-export function asset(url = "") {
+export function asset(ctx: Context, url = "") {
   if (staticUrl.protocol === "file:") {
-    return posix.join(basePath, url);
+    return posix.join(ctx.var.options.basePath, url);
```

Comment:
Passing context through the various places that `asset` or `getPath` are called eventually boils down to having the ability to use the `basePath` that the CMS was initialized with.